services/cloudapi/cloudapinic/nic.go

```diff
@@ -0,0 +1,270 @@
+package cloudapinic
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	ionoscloud "github.com/ionos-cloud/sdk-go/v6"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cloudapi"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+)
+
+type Service struct {
+	Client *ionoscloud.APIClient
+	Meta   interface{}
+	D      *schema.ResourceData
+}
+
+func (fs *Service) Get(ctx context.Context, datacenterID, serverID string, depth int32) ([]ionoscloud.Nic, error) {
+	nics, apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsGet(ctx, datacenterID, serverID).Depth(depth).Execute()
```

Comment:
@cristiGuranIonos it's nice that you included the `depth` here, maybe we should to the same in the future for all other services.